apis/certificatemanager/v1alpha1/certificateauthority_types.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Tag represents user-provided metadata that can be associated
+type Tag struct {
+
+	// The key name that can be used to look up or retrieve the associated value.
+	Key string `json:"key"`
+
+	// The value associated with this tag.
+	// +optional
+	Value string `json:"value,omitempty"`
+}
+
+// CertificateAuthoritySpec defines the desired state of CertificateAuthority
+type CertificateAuthoritySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  CertificateAuthorityParameters `json:"forProvider"`
+}
+
+// CertificateAuthorityExternalStatus keeps the state of external resource
+type CertificateAuthorityExternalStatus struct {
+	// String that contains the ARN of the issued certificate Authority
+	CertificateAuthorityArn string `json:"certificateArn"`
+	RenewalPermission       bool   `json:"renewalPermission"`
+}
+
+// An CertificateAuthorityStatus represents the observed state of an CertificateAuthority manager.
+type CertificateAuthorityStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     CertificateAuthorityExternalStatus `json:"atProvider"`
+}
+
+// CertificateAuthorityParameters defines the desired state of an AWS CertificateAuthority.
+type CertificateAuthorityParameters struct {
+	// Type of the certificate authority
+	Type string `json:"type"`
+
+	// Status of the certificate authority
+	// +optional
+	Status string `json:"status"`
```

Comment:
Is this configurable? It seems that `CertificateAuthorityParameters` may be including fields from multiple AWS API types as well.

Reply:
https://docs.aws.amazon.com/sdk-for-go/v2/api/service/acmpca/#CreateCertificateAuthorityInput

Reply:
Yes, this has fields from multiple AWS types as this will deals CertificateAuthority and CertificateAuthorityRenerwalPermission(only field is CertificateRenewalPermissionAllow).